libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteController.kt

```diff
@@ -0,0 +1,70 @@
+package com.mapbox.navigation.core.fasterroute
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.navigation.base.extensions.ifNonNull
+import com.mapbox.navigation.core.directions.session.DirectionsSession
+import com.mapbox.navigation.core.directions.session.RoutesRequestCallback
+import com.mapbox.navigation.core.trip.session.TripSession
+import com.mapbox.navigation.utils.timer.MapboxTimer
+import java.util.concurrent.CopyOnWriteArrayList
+
+internal class FasterRouteController(
+    private val directionsSession: DirectionsSession,
+    private val tripSession: TripSession
+) {
+    private val fasterRouteTimer = MapboxTimer()
+    private val fasterRouteObservers = CopyOnWriteArrayList<FasterRouteObserver>()
```

Comment:
I think one class will have only one instance of `FasterRouteTimer`; hence does this need to be a list?

Reply:
agree with this. but also out of scope of this refactor because many of our interfaces support multiple subscribers
https://github.com/mapbox/mapbox-navigation-android/blob/d08f1e5c3a0da669167a7ed2351a2ff21f974fc7/examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt#L379-L383

would prefer to separate that refactor

Reply:
Other interfaces support multiple subscribers because a client can do his own implementation along with `MapboxNavigation` doing one itself. Whereas it is different for `FasterRoute` because only `MapboxNavigation` can register for it, whereas others will observe it. There is no need to have a list of subscribers.

Reply:
Ok I was refactoring a few things, and considered this part out of scope. Do you want me to make it part of this refactor or make it a separate pull request?

Reply:
I would prefer to do it in the same PR given the fact it's just a couple of line changes.

Reply:
Ok cool, pushed an update ❤️